src/Phrases/Persistance/Memory.php

```diff
@@ -2,12 +2,22 @@
 
 namespace Phrases\Persistance;
 
+use Phrases\Entity\Phrase;
+
 class Memory implements RepositoryInterface
 {
     protected $list = array();
 
     public function __construct(array $list)
     {
+        foreach ($list as $phrase) {
+            if ($phrase instanceof Phrase) {
+                continue;
+            }
+
+            $message = 'Phrase list should contain only Phrase entities.';
+            throw new \InvalidArgumentException($message);
```

Comment:
We can avoid this one and craete a new strucure only accept `Phrase`, maybe a `PhraseCollection`.